Frameworks/Security/GenericPasswordItemHandler.mm

```diff
@@ -589,16 +558,14 @@ - (OSStatus)remove:(NSDictionary*)queryDictionary {
         return errSecParam;
     }
 
-    // TODO: Error handling when projections support it.
-    NSArray* credArray = [_vault retrieveAll];
+    WFC::IVectorView<PasswordCredential> credArray = _vault.RetrieveAll();
 
     // Documentation for remove states that it by default removes all matching rows while documentation
     // for query defaults to a single item. Go figure.
-    std::vector<StrongId<WSCPasswordCredentialWrapper>> credentialsToRemove = FindMatchingCredentials(queryDictionary, credArray, true);
+    std::vector<StrongId<PasswordCredentialWrapper>> credentialsToRemove = FindMatchingCredentials(queryDictionary, credArray, true);
 
     for (const auto& credential : credentialsToRemove) {
-        // TODO: Error handling when projections support it.
```

Comment:
Yeah I think we should transform to NSException where reasonable. WinRT is getting better at only throwing where exceptional but we probably want to try to do the nice thing where possible.